### PR TITLE
tests: settings: Test setting FS back-end using Littlefs

### PR DESCRIPTION
**Pull request by de-nordic**
The commit ports settings tests that use NFFS to use Littlefs.

GH Issue #18341

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
fcb and nvs tests CMakeList are not prepared to have FS targeted files in the src/ directory

**Comment by nvlsianpu** [comment]
maybe move FC common files under ./fs/src

File: tests/subsys/settings/littlefs/src/settings_test.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SETTINGS_TEST_FCB_H
```
Comment (nvlsianpu) [inline tests/subsys/settings/littlefs/src/settings_test.h]: ~~FCB~~

File: tests/subsys/settings/littlefs/src/settings_test_littlefs.c [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+void test_config_empty_file(void);
+void test_config_small_file(void);
+void test_config_multiple_in_file(void);
+void test_config_save_in_file(void);
+void test_config_save_one_file(void);
+void test_config_compress_file(void);
+
+void test_main(void)
+{
+	ztest_test_suite(test_config_fcb,
```
Comment (nvlsianpu) [inline tests/subsys/settings/littlefs/src/settings_test_littlefs.c]: ~~fcb~~

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: tests/subsys/settings/littlefs/src/settings_test_littlefs.c (hunk 2 above)
Comment (de-nordic) [reply tests/subsys/settings/littlefs/src/settings_test_littlefs.c]: OK

File: tests/subsys/settings/littlefs/src/settings_test.h (hunk 1 above)
Comment (de-nordic) [reply tests/subsys/settings/littlefs/src/settings_test.h]: OK

File: tests/subsys/settings/littlefs/src/settings_setup_littlefs.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	rc = flash_area_open(DT_FLASH_AREA_LITTLEFS_DEV_ID, &fap);
+	zassert_true(rc == 0, "opening flash area for erase [%d]\n", rc);
+
+	flash_area_erase(fap, fap->fa_off, fap->fa_size);
+	zassert_true(rc == 0, "erasing flash area [%d]\n", rc);
+
+	rc = fs_mount(&littlefs_mnt);
+	zassert_true(rc == 0, "mounting littlefs [%d]\n", rc);
+
+	rc = fs_unlink(TEST_CONFIG_DIR);
```
Comment (nvlsianpu) [inline tests/subsys/settings/littlefs/src/settings_setup_littlefs.c]: probably unneeded directory unlink - after the flash area was erased a few line above

File: tests/subsys/settings/nffs/raw/CMakeLists.txt [hunk 4]
```diff
@@ -5,6 +5,7 @@ include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
 project(NONE)
 
 add_subdirectory(../src nffs_test_bindir)
+
```
Comment (nvlsianpu) [inline tests/subsys/settings/nffs/raw/CMakeLists.txt]: add_subdirectory(../../fs/src settings_test_fs_bindir)

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: tests/subsys/settings/littlefs/src/settings_setup_littlefs.c (hunk 3 above)
Comment (de-nordic) [reply tests/subsys/settings/littlefs/src/settings_setup_littlefs.c]: Yes, it is mistake.

File: tests/subsys/settings/nffs/raw/CMakeLists.txt (hunk 4 above)
Comment (de-nordic) [reply tests/subsys/settings/nffs/raw/CMakeLists.txt]: I do not think it is needed. raw only builds ../src, contents. ./src contains CMakeList that specifies this settings_test_fs_bindir.

File: tests/subsys/settings/littlefs/src/settings_setup_littlefs.c (hunk 3 above)
Comment (de-nordic) [reply tests/subsys/settings/littlefs/src/settings_setup_littlefs.c]: Done

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
Works well.

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
cc @carlescufi

**Review by pabigot — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by rakons** [comment]
General question - why all the files was moved from nffs to fs? Did we remove tests for nffs or is is some common test for all fs-es.

**Comment by de-nordic** [comment]
> 
> 
> General question - why all the files was moved from nffs to fs? Did we remove tests for nffs or is is some common test for all fs-es.

Moved source files are common for nffs and littlefs tests. They may be considered common for all fs themed tests.

**Comment by rakons** [comment]
Ok. Thanks.

**Pull request merged**